main.go

```diff
@@ -172,116 +142,125 @@ func main() {
 
 	directCfg := configuration.NewCapsuleConfiguration(ctx, directClient, configurationName)
 
-	ca, err := clientset.CoreV1().Secrets(namespace).Get(ctx, directCfg.CASecretName(), metav1.GetOptions{})
-	if err != nil {
-		setupLog.Error(err, "unable to get Capsule CA secret")
+	tlsReconciler := &tlscontroller.Reconciler{
+		Client:        directClient,
```

Comment:
Using the `directClient` means we're going to query the API server for each read request.

I don't think it's a big problem, here, just want to have feedback from you too.

Reply:
i don't know actually :( but if you think that could be a problem we can use [DelegatingClient](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/client#NewDelegatingClient) with cacheReader

Reply:
No problem at all, let's see in the future if we have to change it, since TLS shouldn't be changes so frequently.